File: apps/backend-elysia/src/common/response.ts
import { ErrorCode } from "./error-code"

export function okResponse<T>(data: T) {
  return {
    code: ErrorCode.ok,
    message: "ok",
    data,
  }
}

export function failResponse(code: 10001 | 10007, message: string) {
  return {
    code,
    message,
    data: {},
  }
}

File: apps/backend-axum/src/common/response.rs
use serde::Serialize;

#[derive(Serialize)]
pub(crate) struct ApiResponse<T> {
    pub(crate) code: u32,
    pub(crate) message: String,
    pub(crate) data: T,
}

pub(crate) fn success<T>(data: T) -> ApiResponse<T> {
    ApiResponse {
        code: 0,
        message: "ok".to_owned(),
        data,
    }
}
